### PR TITLE
[SYCL-PTX] Add builtins for the relational category

### PR DESCRIPTION
**Pull request by Naghasan**
Remove old forward declared SPIR-V builtin from the relational category.
Update the binding file using the header generator.
Add relevant binding test.
Remove XFAIL from related SYCL tests.

Note there is a dependency with https://github.com/intel/llvm/pull/1828 but can be reviewed independently.

Signed-off-by: Victor Lomuller <victor@codeplay.com>

### CODE REVIEW

File: libclc/utils/gen-libclc-test.py [hunk 1]
```diff
@@ -13,11 +13,30 @@
 
 def ignore_function(fun):
     whitelist = [
+<<<<<<< HEAD
```
Comment (bader) [inline libclc/utils/gen-libclc-test.py:16]: This should go away after rebasing to #1828.

File: libclc/test/binding/ocl/bitselect.cl [hunk 2]
```diff
@@ -0,0 +1,456 @@
+
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// Autogenerated by gen-libclc-test.py
```
Comment (bader) [inline libclc/test/binding/ocl/bitselect.cl:10]: Should we keep auto-generated tests under version control or just the script which generates them?

**Comment by rbegam** [comment]
Hi @Naghasan this is marked as WIP. Do you want me to review it now or wait for the next commit?

**Comment by Naghasan** [comment]
> Hi @Naghasan this is marked as WIP. Do you want me to review it now or wait for the next commit?

@rbegam That can be review, the WIP is only because there is an implicit dependency with #1828 so can't build on its own. Although, #1828 is not required to review this one.

File: libclc/test/binding/ocl/bitselect.cl (hunk 2 above)
Comment (Naghasan) [reply libclc/test/binding/ocl/bitselect.cl:10]: The script takes a while to generate everything, it is only not meant to be constantly rerun as the builtins are not meant to change.

File: libclc/utils/gen-libclc-test.py (hunk 1 above)
Comment (Naghasan) [reply libclc/utils/gen-libclc-test.py]: I fixed this

**Comment by Naghasan** [comment]
@bader @rbegam bumped to sycl tip and fixed the git conflict marker issue

**Comment by bader** [comment]
@Naghasan, #1828 is merged. Could you resolve merge conflicts, please?

**Comment by Naghasan** [comment]
@bader done :)

**Comment by bader** [comment]
@intel/llvm-reviewers-runtime, ping.

**Pull request merged**